hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/repository/EntityRepository.java

```diff
@@ -43,4 +43,7 @@ public interface EntityRepository extends CrudRepository<Entity, Long> {
 
     @Query(value = "select id from entity where evm_address = ?1 and deleted <> true", nativeQuery = true)
     Optional<Long> findAccountIdByAddress(byte[] evmAddress);
+
+    @Query(value = "select balance from entity where evm_address = ?1 and deleted <> true", nativeQuery = true)
```

Comment:
Deleted can be null, in which case <> won't work as you expect and will omit deleted that are `NULL`. Also, only accounts and contracts can have balances so should filter down.
```suggestion
    @Query(value = "select balance from entity where evm_address = ?1 and type in ('ACCOUNT', 'CONTRACT') and coalesce(deleted, false) <> true", nativeQuery = true)
```

Reply:
or `deleted is not false` also works